### PR TITLE
Active asset

### PR DESCRIPTION
**Pull request by Reettaphant**
Add a method to mark an asset as active after it has been encoding by youtube. Depends on this media atom maker endpoint: https://github.com/guardian/media-atom-maker/pull/104

@akash1810 @clloyd

### CODE REVIEW

File: CDS/js/HMACRequest.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+var reqwest = require('reqwest');
+var hmac = require('./hmac');
+
+module.exports = {
+  makeRequest: function(connection, date, uri, urlBase, method, data) {
+
+    return hmac.makeHMACToken(connection, date, uri)
+    .then(token => {
+
+      let requestBody = {
```
Comment (akash1810) [inline CDS/js/HMACRequest.js]: `const`?

File: CDS/js/HMACRequest.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+var reqwest = require('reqwest');
```
Comment (akash1810) [inline CDS/js/HMACRequest.js]: qq - why not use `import` syntax here?

File: CDS/js/media-atom-lib.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
-var reqwest = require('reqwest');
 var Promise = require('promise');
 var datastore = require('./Datastore');
-var hmac = require('./hmac');
+var HMACRequest = require('./HMACRequest');
+var reqwest = require('reqwest');
```
Comment (akash1810) [inline CDS/js/media-atom-lib.js]: doesn't look like we're using this anymore?

File: CDS/js/media-atom-lib.js [hunk 4]
```diff
@@ -41,32 +42,76 @@ function fetchMetadata(connection) {
...
+
+    function makeActive() {
+
+      const date = (new Date()).toUTCString();
+      return HMACRequest.makeRequest(connection, date, uri, urlBase, 'POST', data)
+      .then(response => {
+        return response;
+      })
+      .catch(error => {
+        if (error.response === 'Asset encoding in process') {
```
Comment (akash1810) [inline CDS/js/media-atom-lib.js]: should we rely on http response code too?

File: CDS/js/HMACRequest.js (hunk 1 above)
Comment (Reettaphant) [inline CDS/js/HMACRequest.js]: It may get an new attribute added on line 22

File: CDS/js/media-atom-lib.js (hunk 4 above)
Comment (Reettaphant) [inline CDS/js/media-atom-lib.js]: Maybe, I'm just sending a generic error code so I'm not sure how useful it is.

File: CDS/js/HMACRequest.js (hunk 2 above)
Comment (Reettaphant) [inline CDS/js/HMACRequest.js]: I don't think there is any other reason not to than just being consistent with the js already here.

Comment (clloyd) [inline CDS/js/HMACRequest.js]: This is a node app `require` is the standard

**Comment by Reettaphant** [comment]
I've now updated this pr and solved the merge conflicts with master @clloyd

File: CDS/scripts/js_utils/media-atom-lib.js [hunk 5]
```diff
@@ -41,34 +41,76 @@ function fetchMetadata(connection) {
...
-                    return response;
-                });
+
+        return HMACRequest.makeRequest(connection, date, uri, urlBase, 'GET')
+        .then(response => {
+            const title = response.data.title;
+            const description = response.data.description;
+            const categoryId = response.data.categoryId;
+
+            return Promise.all([datastore.set(connection, 'meta', 'atom_title', title), datastore.set(connection, 'meta', 'atom_description', description), datastore.set(connection, 'meta', 'atom_category', categoryId)])
```
Comment (fredex42) [inline CDS/scripts/js_utils/media-atom-lib.js]: you could use datastore.setMulti here

**Comment by clloyd** [comment]
@fredex42 I think I've fixed the naming in #34 These should get changes when master is merged in

**Comment by clloyd** [comment]
@fredex42 is this ok to merge once the merge conflicts are fixed

**Comment by Reettaphant** [comment]
@fredex42 I've fixed the add asset script name and solved the merge conflicts now.

**Pull request merged**